src/TwigComponent/src/ComponentFactory.php

```diff
@@ -47,7 +47,7 @@ public function metadataFor(string $name): ComponentMetadata
     /**
      * Creates the component and "mounts" it with the passed data.
      */
-    public function create(string $name, array $data = []): object
+    public function create(string $name, array $data = []): MountedComponent
```

Comment:
This is one of the spots where adding this feature makes things feel bigger and less "pure". However, this has grown on me. I like that we have a method where you pass in the string component name + array of data, and you get back the (A) the component object and (B) the extra stuff as attributes. I think that feels good.